omod/src/test/java/org/openmrs/module/idgen/rest/controller/AutoGenerationOptionControllerTest.java

```diff
@@ -0,0 +1,206 @@
+		Object autogenerationOptionUuid = PropertyUtils.getProperty(response, "uuid");
+		AutoGenerationOption newAutoGenerationOption = identifierSourceService
+		        .getAutoGenerationOption(Integer.parseInt(autogenerationOptionUuid.toString()));
+		assertEquals(SOURCE, newAutoGenerationOption.getSource().getUuid());
+		assertNull(newAutoGenerationOption.getLocation());
+		assertEquals(IDENTIFIER_TYPE_UUID, newAutoGenerationOption.getIdentifierType().getUuid());
+		assertEquals(true, newAutoGenerationOption.isAutomaticGenerationEnabled());
+		assertEquals(true, newAutoGenerationOption.isManualEntryEnabled());
+	}
+	
+	@Test(expected = ValidationException.class)
+	public void shouldNotCreateNewAutoGenerationOptionWithNullParameters() throws Exception {
+		SimpleObject autoGenerationOption = new SimpleObject();
+		autoGenerationOption.add("identifierType", null);
+		autoGenerationOption.add("location", null);
+		autoGenerationOption.add("source", null);
+		autoGenerationOption.add("automaticGenerationEnabled", null);
+		autoGenerationOption.add("manualEntryEnabled", null);
+		String json = new ObjectMapper().writeValueAsString(autoGenerationOption);
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI());
+		req.setContent(json.getBytes());
+		handle(req);
+	}
+	
+	@Test
+	public void shouldEditAnAutoGenerationOption() throws Exception {
+		AutoGenerationOption autogenerationOption = identifierSourceService.getAutoGenerationOption(getId());
+		assertEquals(LOCATION, autogenerationOption.getLocation().getUuid());
+		assertEquals(SOURCE, autogenerationOption.getSource().getUuid());
+		assertTrue(autogenerationOption.isAutomaticGenerationEnabled());
+		assertTrue(autogenerationOption.isManualEntryEnabled());
+		SimpleObject autoGenerationOption = new SimpleObject();
+		autoGenerationOption.add("location", NEW_LOCATION);
+		autoGenerationOption.add("source", NEW_SOURCE);
+		autoGenerationOption.add("manualEntryEnabled", false);
+		autoGenerationOption.add("automaticGenerationEnabled", false);
+		String json = new ObjectMapper().writeValueAsString(autoGenerationOption);
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI() + "/" + getUuid());
+		req.setContent(json.getBytes());
+		handle(req);
+		AutoGenerationOption updatedAutogenerationOption = identifierSourceService.getAutoGenerationOption(getId());
+		assertEquals(NEW_LOCATION, updatedAutogenerationOption.getLocation().getUuid());
+		assertEquals(NEW_SOURCE, updatedAutogenerationOption.getSource().getUuid());
+		assertFalse(updatedAutogenerationOption.isAutomaticGenerationEnabled());
+		assertFalse(updatedAutogenerationOption.isManualEntryEnabled());
+	}
+	
+	@Test(expected = ResourceDoesNotSupportOperationException.class)
+	public void shouldNotEditAnAutoGenerationOptionProvidedWithAllNullParams() throws Exception {
+		SimpleObject autoGenerationOption = new SimpleObject();
+		autoGenerationOption.add("location", null);
+		autoGenerationOption.add("source", null);
```

Comment:
For null values, we just do not supply those properties in the autoGenerationOption